packages/cloud-cognitive/src/components/SidePanel/utils.js

```diff
@@ -0,0 +1,5 @@
+export const changeArrayPosition = (arr, originalPosition, newPosition) => {
+  let cutOut = arr.splice(originalPosition, 1)[0];
```

Comment:
we should try not to mutate data whenever possible and unfortunately `splice` is not immutable. i'd recommend just adding this one additional line to create a new array
```
const changeArrayPosition = (arr, originalPosition, newPosition) => {
  const newArr = [...arr];
  const cutOut = arr.splice(originalPosition, 1)[0];
  arr.splice(newPosition, 0, cutOut);
  return arr;
};
```
and now the original data is still intact.

Reply:
Fixed [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/551/commits/e99dd8850916e6c6d11bee807590610ea2a23e8e)